x-pack/plugins/cases/common/api/cases/comment.ts

```diff
@@ -68,22 +70,69 @@ export const ActionsCommentRequestRt = rt.type({
   owner: rt.string,
 });
 
+export enum ExternalReferenceStorageType {
+  so = 'so',
+  doc = 'doc',
```

Comment:
@jonathan-buttner I changed the `other` to `doc` to have control of the supported types. `doc` stands for a document in ES. Let me know if the naming is not good.